Client/game_sa/CColModelSA.h

```diff
@@ -58,19 +63,123 @@ typedef struct
     char  name[0x18];
 } ColModelFileHeader;
 
+typedef struct
+{
+    CVector m_vecStart;
+    float   m_fStartSize;
+    CVector m_vecEnd;
+    float   m_fEndSize;
+} CColLine;
+
+typedef struct
+{
+    CVector       m_vecStart;
+    float         m_fStartRadius;
+    unsigned char m_nMaterial;
+    unsigned char m_nPiece;
+    unsigned char m_nLighting;
+    char          _pad13;
+    CVector       m_vecEnd;
+    float         m_fEndRadius;
+} CColDisk;
+
+typedef struct
+{
+    signed __int16 x;
+    signed __int16 y;
+    signed __int16 z;
+    CVector        getVector() const { return CVector(x * 0.0078125f, y * 0.0078125f, z * 0.0078125f); }
+    void           setVector(CVector vec)
+    {
+        x = static_cast<signed __int16>(vec.fX * 128);
+        y = static_cast<signed __int16>(vec.fY * 128);
+        z = static_cast<signed __int16>(vec.fZ * 128);
+    }
+} CompressedVector;
+
 typedef struct
 {
     WORD                 numColSpheres;
     WORD                 numColBoxes;
     WORD                 numColTriangles;
     BYTE                 ucNumWheels;
-    BYTE                 pad3;
+    BYTE                 m_nFlags;
     CColSphereSA*        pColSpheres;
     CColBoxSA*           pColBoxes;
     void*                pSuspensionLines;
-    void*                pUnknown;
+    CompressedVector*    pVertices;
     CColTriangleSA*      pColTriangles;
     CColTrianglePlaneSA* pColTrianglePlanes;
+    unsigned int         m_nNumShadowTriangles;
+    unsigned int         m_nNumShadowVertices;
+    CompressedVector*    m_pShadowVertices;
+    CColTriangleSA*      m_pShadowTriangles;
+
+    std::map<ushort, CompressedVector> getAllVertices() const
+    {
+        std::map<ushort, CompressedVector> vertices;
+        for (uint i = 0; numColTriangles > i; i++)
+        {
+            vertices[pColTriangles[i].vertex[0]] = pVertices[pColTriangles[i].vertex[0]];
+            vertices[pColTriangles[i].vertex[1]] = pVertices[pColTriangles[i].vertex[1]];
+            vertices[pColTriangles[i].vertex[2]] = pVertices[pColTriangles[i].vertex[2]];
+        }
+        return vertices;
+    }
```

Comment:
Instead of returning ` std::map<ushort, CompressedVector>`, it's better to return `std::map<ushort, CompressedVector> &` for good performance . If you find it hard to refactor the return type, the least you can is to pass the map container by reference as a parameter.
```
void getAllVertices(std::map<ushort, CompressedVector>& vertices) const
{
    for (uint i = 0; numColTriangles > i; i++)
    {
        vertices[pColTriangles[i].vertex[0]] = pVertices[pColTriangles[i].vertex[0]];
        vertices[pColTriangles[i].vertex[1]] = pVertices[pColTriangles[i].vertex[1]];
        vertices[pColTriangles[i].vertex[2]] = pVertices[pColTriangles[i].vertex[2]];
    }
}
```

Reply:
Im pretty sure it gets RVO'd.